ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -528,6 +718,156 @@ def test_rebuild_membership_hosts_group(self, automember_hostgroup, user1,
+                                             hostgroup1):
+        hostgroup1.ensure_exists()
+        command = automember_hostgroup.add_condition(
+            key=u'fqdn', type=u'badtype',
+            inclusiveregex="qa[0-9]+.example.com")
+        with raises_exact(errors.NotFound(
+                reason=u'invalid \'type\': must be one '
+                       u'of \'group\', \'hostgroup\'')):
+            command()
+
+    def test_create_exclusive_with_hostgroup(self, automember_hostgroup,
+                                             hostgroup1):
+        hostgroup1.ensure_exists()
+        command = automember_hostgroup.add_condition(
+            key=u'fqdn', type=u'badtype',
+            exclusiveregex="eng[0-9]+.example.com")
+        with raises_exact(errors.NotFound(
+                reason=u'invalid \'type\': must be one '
+                       u'of \'group\', \'hostgroup\'')):
+            command()
+
+
+class TestAutomemberAddConditionNegativeBadkey(XMLRPC_test):
+
+    def test_inclusive_with_group_invalidkey(self, automember_group, group1):
+        group1.check_add_condition_negative()
+        command = automember_group.add_condition(
+            key=u'badkey', type=u'group',
+            inclusiveregex=[group_include_regex])
+        with raises_exact(errors.NotFound(
+                reason=u'badkey is not a valid attribute.')):
+            command()
+
+    def test_exclusive_with_group_invalidkey(self, automember_group, group1):
+        group1.check_add_condition_negative()
+        command = automember_group.add_condition(key=u'badkey',
+                                                 type=u'group',
+                                                 exclusiveregex=[u'mjohn'])
+        with raises_exact(errors.NotFound(
+                reason=u'badkey is not a valid attribute.')):
+            command()
+
+    def test_inclusive_with_hostgroup_invalidkey(
+            self, automember_hostgroup, hostgroup1):
+        hostgroup1.check_add_condition_negative()
+        command = automember_hostgroup.add_condition(
+            key=u'badkey', type=u'hostgroup',
+            inclusiveregex="qa[0-9]+.example.com")
+        with raises_exact(errors.NotFound(
+                reason=u'badkey is not a valid attribute.')):
+            command()
+
+    def test_exclusive_with_hostgroup_invalidkey(
+            self, automember_hostgroup, hostgroup1):
+        hostgroup1.check_add_condition_negative()
+        command = automember_hostgroup.add_condition(
+            key=u'badkey', type=u'hostgroup',
+            exclusiveregex="eng[0-9]+.example.com")
+        with raises_exact(errors.NotFound(
+                reason=u'badkey is not a valid attribute.')):
+            command()
+
+
+class TestAutomemberAddConditionNegativeBadregextype(XMLRPC_test):
+
+    def test_create_inclusive_with_group(self, automember_group, group1):
+        group1.check_add_condition()
```

Comment:
Replace the above line with group1.ensure_exists() and automember_group.ensure_exists()